Sources/_RegexParser/Regex/Parse/Parse.swift

```diff
@@ -289,12 +289,16 @@ extension Parser {
   /// Apply the syntax options of a given matching option sequence to the
   /// current set of options.
   private mutating func applySyntaxOptions(
-    of opts: AST.MatchingOptionSequence
-  ) {
+    of opts: AST.MatchingOptionSequence, isScoped: Bool
+  ) throws {
     func mapOption(_ option: SyntaxOptions,
                    _ pred: (AST.MatchingOption) -> Bool) {
       if opts.resetsCurrentOptions {
-        context.syntax.remove(option)
+        // Extended syntax is never reset in a multi-line literal.
+        if !(option == .extendedSyntax &&
+             context.syntax.contains(.multilineLiteral)) {
+          context.syntax.remove(option)
+        }
```

Comment:
@milseman Just to double check, `(?^:...)` shouldn't unset extended syntax in a multi-line literal? This is consistent with the behavior of `(?^)` in a multi-line literal, as otherwise we'd need to ban it outright.

Reply:
If we can, I think `(?^)` meaning reset to default (and the default is determined by literal / run time string vs multi-line literal) would be best

Reply:
We talked about this offline and decided to ban `(?^)` outright as it's specific to PCRE and is defined to be equivalent to `(?-imnsx)`